// saga/dist/dispatcher.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var DispatcherClass = /** @class */ (function () {
    function DispatcherClass() {
        this.stores = {};
        this._isDispatching = false;
    }
    DispatcherClass.prototype.register = function (name, callback) {
        this.stores[name] = {
            callback: callback,
            isHandled: false,
            isPending: false
        };
    };
    DispatcherClass.prototype.unregister = function (name) {
        this.stores[name] = undefined;
    };
    DispatcherClass.prototype.waitFor = function (names) {
        if (!this.isDispatching) {
            throw new Error("Dispatcher.waitFor(...): Must be invoked while dispatching.");
        }
        for (var _i = 0, names_1 = names; _i < names_1.length; _i++) {
            var name_1 = names_1[_i];
            var store = this.stores[name_1];
            if (store == null) {
                throw new Error("Dispatcher.waitFor(...): Store \"" + name_1 + "\" must be initialized first.");
            }
            if (store.isPending) {
                throw new Error("Dispatcher.waitFor(...): Circular dependency detected while waiting for " + name_1 + ".");
            }
            this.invokeCallback(store);
        }
    };
    DispatcherClass.prototype.dispatch = function (payload) {
        if (this._isDispatching) {
            throw new Error("Dispatch.dispatch(...): Cannot dispatch in the middle of a dispatch.");
        }
        this.startDispatching(payload);
        try {
            for (var _i = 0, _a = Object.keys(this.stores); _i < _a.length; _i++) {
                var name_2 = _a[_i];
                var store = this.stores[name_2];
                if (store == null || store.isPending) {
                    continue;
                }
                this.invokeCallback(store);
            }
        }
        finally {
            this.stopDispatching();
        }
    };
    Object.defineProperty(DispatcherClass.prototype, "isDispatching", {
        get: function () {
            return this._isDispatching;
        },
        enumerable: true,
        configurable: true
    });
    DispatcherClass.prototype.invokeCallback = function (store) {
        if (this.pendingPayload == null) {
            return;
        }
        store.isPending = true;
        store.callback(this.pendingPayload);
        store.isHandled = true;
    };
    DispatcherClass.prototype.startDispatching = function (payload) {
        for (var _i = 0, _a = Object.keys(this.stores); _i < _a.length; _i++) {
            var name_3 = _a[_i];
            var store = this.stores[name_3];
            if (store == null) {
                continue;
            }
            store.isPending = false;
            store.isHandled = false;
        }
        this.pendingPayload = payload;
        this._isDispatching = true;
    };
    DispatcherClass.prototype.stopDispatching = function () {
        this.pendingPayload = undefined;
        this._isDispatching = false;
    };
    return DispatcherClass;
}());
function createDispatcher() {
    return new DispatcherClass();
}
exports.createDispatcher = createDispatcher;
