// saga/dist/emitter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var TinyEmitter = /** @class */ (function () {
    function TinyEmitter() {
    }
    TinyEmitter.prototype.addListener = function (handler) {
        var _this = this;
        if (this.registry == null) {
            this.registry = handler;
        }
        else if (Array.isArray(this.registry)) {
            this.registry.push(handler);
        }
        else {
            this.registry = [this.registry, handler];
        }
        return function () { return _this.removeListener(handler); };
    };
    TinyEmitter.prototype.removeListener = function (handler) {
        if (this.registry == null) {
            return;
        }
        if (Array.isArray(this.registry)) {
            var nextArray = [];
            for (var i = 0; i < this.registry.length; i++) {
                var callback = this.registry[i];
                if (callback !== handler) {
                    nextArray.push(callback);
                }
            }
            this.registry = nextArray;
        }
        else {
            this.registry = undefined;
        }
    };
    TinyEmitter.prototype.emit = function () {
        var payload = [];
        for (var _i = 0; _i < arguments.length; _i++) {
            payload[_i] = arguments[_i];
        }
        var _a;
        if (this.registry == null) {
            return;
        }
        if (Array.isArray(this.registry)) {
            for (var i = 0; i < this.registry.length; i++) {
                (_a = this.registry)[i].apply(_a, payload);
            }
        }
        else {
            this.registry.apply(this, payload);
        }
    };
    TinyEmitter.prototype.getCount = function () {
        if (this.registry == null) {
            return 0;
        }
        if (Array.isArray(this.registry)) {
            return this.registry.length;
        }
        else {
            return 1;
        }
    };
    return TinyEmitter;
}());
exports.TinyEmitter = TinyEmitter;
