// saga/dist/store.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
var emitter_1 = require("./emitter");
var StoreClass = /** @class */ (function () {
    function StoreClass(name, initialState, dispatcher, reducer, areEqual) {
        var _this = this;
        this.name = name;
        this.initialState = initialState;
        this.dispatcher = dispatcher;
        this.reducer = reducer;
        this.areEqual = areEqual;
        this.state = tslib_1.__assign({}, this.initialState);
        this.hasStoreChanged = false;
        this.emitter = new emitter_1.TinyEmitter();
        this.onDispatch = function (payload) {
            _this.hasStoreChanged = false;
            var currentState = _this.state;
            var nextState = _this.reducer(currentState, payload);
            if (!_this.areEqual(currentState, nextState)) {
                _this.state = nextState;
                _this.hasStoreChanged = true;
                _this.emitter.emit();
            }
        };
        this.dispatcher.register(this.name, this.onDispatch);
    }
    StoreClass.prototype.getName = function () {
        return this.name;
    };
    StoreClass.prototype.getState = function () {
        return this.state;
    };
    StoreClass.prototype.getDispatcher = function () {
        return this.dispatcher;
    };
    StoreClass.prototype.hasChanged = function () {
        return this.hasStoreChanged;
    };
    StoreClass.prototype.subscribe = function (callback) {
        return this.emitter.addListener(callback);
    };
    StoreClass.prototype.unsubscribe = function (callback) {
        this.emitter.removeListener(callback);
    };
    StoreClass.prototype.getSubscribersCount = function () {
        return this.emitter.getCount();
    };
    return StoreClass;
}());
function createStore(options) {
    var areEqual = options.areEqual != null ? options.areEqual : function (state, nextState) { return state === nextState; };
    return new StoreClass(options.name, options.initialState, options.dispatcher, options.reducer, areEqual);
}
exports.createStore = createStore;
