// es-html-parser/dist/tokenizer/handlers/attribute-key.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.parse = void 0;
const constants_1 = require("../../constants");
const token_types_1 = require("../../constants/token-types");
const utils_1 = require("../../utils");
function parse(chars, state, tokens) {
    if (isKeyBreak(chars)) {
        return parseKeyEnd(state, tokens);
    }
    state.accumulatedContent += state.decisionBuffer;
    state.decisionBuffer = "";
    state.caretPosition++;
}
exports.parse = parse;
function isKeyBreak(chars) {
    return chars === "=" || chars === "/" || chars === ">" || (0, utils_1.isWhitespace)(chars);
}
function parseKeyEnd(state, tokens) {
    const position = (0, utils_1.calculateTokenPosition)(state, { keepBuffer: false });
    tokens.push({
        type: token_types_1.TokenTypes.AttributeKey,
        value: state.accumulatedContent,
        range: position.range,
        loc: position.loc,
    });
    state.accumulatedContent = "";
    state.decisionBuffer = "";
    state.currentContext = constants_1.TokenizerContextTypes.Attributes;
}
