// es-html-parser/dist/constants/construct-tree-context-types.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConstructTreeContextTypes = void 0;
var ConstructTreeContextTypes;
(function (ConstructTreeContextTypes) {
    ConstructTreeContextTypes["TagContent"] = "TagContent";
    ConstructTreeContextTypes["Tag"] = "Tag";
    ConstructTreeContextTypes["TagName"] = "TagName";
    ConstructTreeContextTypes["Attributes"] = "Attributes";
    ConstructTreeContextTypes["Attribute"] = "Attribute";
    ConstructTreeContextTypes["AttributeValue"] = "AttributeValue";
    ConstructTreeContextTypes["Comment"] = "Comment";
    ConstructTreeContextTypes["Doctype"] = "Doctype";
    ConstructTreeContextTypes["DoctypeAttributes"] = "DoctypeAttributes";
    ConstructTreeContextTypes["DoctypeAttribute"] = "DoctypeAttribute";
    ConstructTreeContextTypes["ScriptTag"] = "ScriptTag";
    ConstructTreeContextTypes["StyleTag"] = "StyleTag";
})(ConstructTreeContextTypes = exports.ConstructTreeContextTypes || (exports.ConstructTreeContextTypes = {}));
