// es-html-parser/dist/tokenizer/handlers/doctype-start.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.parse = void 0;
const constants_1 = require("../../constants");
const utils_1 = require("../../utils");
function parse(chars, state, tokens) {
    if ((0, utils_1.isWhitespace)(chars)) {
        return parseWhitespace(state, tokens);
    }
    if (chars === ">") {
        return parseClosingCornerBrace(state, tokens);
    }
    state.decisionBuffer = "";
    state.caretPosition++;
}
exports.parse = parse;
function generateDoctypeOpenToken(state) {
    const position = (0, utils_1.calculateTokenPosition)(state, { keepBuffer: false });
    return {
        type: constants_1.TokenTypes.DoctypeOpen,
        value: state.accumulatedContent,
        range: position.range,
        loc: position.loc,
    };
}
function parseWhitespace(state, tokens) {
    tokens.push(generateDoctypeOpenToken(state));
    state.accumulatedContent = "";
    state.decisionBuffer = "";
    state.currentContext = constants_1.TokenizerContextTypes.DoctypeAttributes;
}
function parseClosingCornerBrace(state, tokens) {
    tokens.push(generateDoctypeOpenToken(state));
    state.accumulatedContent = "";
    state.decisionBuffer = "";
    state.currentContext = constants_1.TokenizerContextTypes.DoctypeClose;
}
