// es-html-parser/dist/tokenizer/handlers/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.noop = exports.styleTagContent = exports.scriptTagContent = exports.openTagStart = exports.openTagEnd = exports.DoctypeOpen = exports.DoctypeClose = exports.doctypeAttributes = exports.doctypeAttributeWrapped = exports.doctypeAttributeBare = exports.data = exports.commentContent = exports.closeTag = exports.attributes = exports.attributeValue = exports.attributeValueWrapped = exports.attributeValueBare = exports.attributeKey = void 0;
exports.attributeKey = __importStar(require("./attribute-key"));
exports.attributeValueBare = __importStar(require("./attribute-value-bare"));
exports.attributeValueWrapped = __importStar(require("./attribute-value-wrapped"));
exports.attributeValue = __importStar(require("./attribute-value"));
exports.attributes = __importStar(require("./attributes"));
exports.closeTag = __importStar(require("./close-tag"));
exports.commentContent = __importStar(require("./comment-content"));
exports.data = __importStar(require("./data"));
exports.doctypeAttributeBare = __importStar(require("./doctype-attribute-bare"));
exports.doctypeAttributeWrapped = __importStar(require("./doctype-attribute-wrapped"));
exports.doctypeAttributes = __importStar(require("./doctype-attributes"));
exports.DoctypeClose = __importStar(require("./doctype-end"));
exports.DoctypeOpen = __importStar(require("./doctype-start"));
exports.openTagEnd = __importStar(require("./open-tag-end"));
exports.openTagStart = __importStar(require("./open-tag-start"));
exports.scriptTagContent = __importStar(require("./script-tag-content"));
exports.styleTagContent = __importStar(require("./style-tag-content"));
exports.noop = {
    parse: () => void 0,
};
