// es-html-parser/dist/constants/token-types.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TokenTypes = void 0;
var TokenTypes;
(function (TokenTypes) {
    TokenTypes["Text"] = "Text";
    TokenTypes["OpenTagStart"] = "OpenTagStart";
    TokenTypes["OpenTagEnd"] = "OpenTagEnd";
    TokenTypes["CloseTag"] = "CloseTag";
    TokenTypes["AttributeKey"] = "AttributeKey";
    TokenTypes["AttributeAssignment"] = "AttributeAssignment";
    TokenTypes["AttributeValueWrapperStart"] = "AttributeValueWrapperStart";
    TokenTypes["AttributeValue"] = "AttributeValue";
    TokenTypes["AttributeValueWrapperEnd"] = "AttributeValueWrapperEnd";
    TokenTypes["DoctypeOpen"] = "DoctypeOpen";
    TokenTypes["DoctypeAttributeValue"] = "DoctypeAttributeValue";
    TokenTypes["DoctypeAttributeWrapperStart"] = "DoctypeAttributeWrapperStart";
    TokenTypes["DoctypeAttributeWrapperEnd"] = "DoctypeAttributeWrapperEnd";
    TokenTypes["DoctypeClose"] = "DoctypeClose";
    TokenTypes["CommentOpen"] = "CommentOpen";
    TokenTypes["CommentContent"] = "CommentContent";
    TokenTypes["CommentClose"] = "CommentClose";
    TokenTypes["OpenScriptTagStart"] = "OpenScriptTagStart";
    TokenTypes["OpenScriptTagEnd"] = "OpenScriptTagEnd";
    TokenTypes["ScriptTagContent"] = "ScriptTagContent";
    TokenTypes["CloseScriptTag"] = "CloseScriptTag";
    TokenTypes["OpenStyleTagStart"] = "OpenStyleTagStart";
    TokenTypes["OpenStyleTagEnd"] = "OpenStyleTagEnd";
    TokenTypes["StyleTagContent"] = "StyleTagContent";
    TokenTypes["CloseStyleTag"] = "CloseStyleTag";
})(TokenTypes = exports.TokenTypes || (exports.TokenTypes = {}));
