// @html-eslint/parser/dist/parser.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.parseForESLint = void 0;
const es_html_parser_1 = require("es-html-parser");
const visitor_keys_1 = require("./visitor-keys");
const traverse_1 = require("./traverse");
function parseForESLint(code) {
    const { ast, tokens } = (0, es_html_parser_1.parse)(code);
    const programNode = {
        type: "Program",
        body: ast.children,
        loc: ast.loc,
        range: ast.range,
        tokens: tokens.filter((token) => token.type !== "CommentContent" &&
            token.type !== "CommentOpen" &&
            token.type !== "CommentClose"),
        comments: [],
    };
    (0, traverse_1.traverse)(programNode, (node) => {
        if (node.type === "CommentContent") {
            programNode.comments.push({
                type: node.type,
                range: node.range,
                loc: node.loc,
                value: node.value,
            });
        }
    });
    return {
        ast: programNode,
        visitorKeys: visitor_keys_1.visitorKeys,
        scopeManager: null,
    };
}
exports.parseForESLint = parseForESLint;
